script.js

```diff
@@ -1,4 +1,115 @@
+var currentGameMode = "waiting for player 1";
+var randomDiceRoll = [];
+
+var rollDiceOne = 0;
+var rollDiceTwo = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var userName = "";
+
+  // how to ensure they are 2 separate numbers and not a summation of both values?
+
+  // Game Mode One: Player 1
+
+  // Insert username
+  if (currentGameMode == "waiting for player 1") {
+    console.log(currentGameMode);
+    currentGameMode = "pick dice for player 1";
+    userName = input;
+    rollDiceOne = rollDice();
+    rollDiceTwo = rollDice();
+    myOutputValue = myOutputValue =
+      "Hello, " +
+      userName +
+      ". " +
+      "You rolled " +
+      Number(rollDiceOne) +
+      " for Dice 1 and " +
+      Number(rollDiceTwo) +
+      " for Dice 2. <br>Please input 1 or 2 to indicate which number you would like to be the first number.";
+  }
+  // Score for Player 1
+  else if (currentGameMode == "pick dice for player 1") {
+    console.log(currentGameMode);
+    if (input == 1) {
+      currentGameMode = "waiting for player 2";
+      myOutputValue =
+        "Your final score is " +
+        Number(String(rollDiceTwo) + String(rollDiceOne)) +
+        " Please enter player 2's name.";
+      randomDiceRoll.push(Number(String(rollDiceTwo) + String(rollDiceOne)));
```

Comment:
You will need to double-check the order of `rollDiceTwo` and `rollDiceOne` if `(input == 1)`. I presume you want to do `rollDiceOne`, then `rollDiceTwo`, right? :P